tests/proptest.rs

```diff
@@ -2,35 +2,39 @@
 
 use std::{collections::BTreeSet as Set, error::Error};
 
+use pubgrub::error::PubGrubError;
+use pubgrub::package::Package;
+use pubgrub::range::Range;
+use pubgrub::report::{DefaultStringReporter, Reporter};
+use pubgrub::solver::{
+    choose_package_with_fewest_versions, resolve, Dependencies, DependencyProvider,
+    OfflineDependencyProvider,
+};
+use pubgrub::version::{NumberVersion, Version};
+
 use proptest::collection::{btree_map, vec};
 use proptest::prelude::*;
 use proptest::sample::Index;
 use proptest::string::string_regex;
 
-use pubgrub::range::Range;
-use pubgrub::solver::{resolve, Dependencies, DependencyProvider, OfflineDependencyProvider};
-use pubgrub::{
-    error::PubGrubError, package::Package, report::DefaultStringReporter, report::Reporter,
-    version::NumberVersion, version::Version,
-};
-
 use crate::sat_dependency_provider::SatResolve;
 
 mod sat_dependency_provider;
 
-/// The same as DP but takes versions from the opposite end:
-/// if DP returns versions from newest to oldest, this returns them from oldest to newest.
+/// The same as [OfflineDependencyProvider] but takes versions from the opposite end:
+/// if [OfflineDependencyProvider] returns versions from newest to oldest, this returns them from oldest to newest.
```

Comment:
Is it still true?

```
        Ok(choose_package_with_fewest_versions(
            v
            |p| self.0.versions(p).into_iter().flatten().cloned(),
        })
            potential_packages,
        ))
```
Does not look like we pick "oldest" versions here anymore :thinking:

Reply:
`versions` currently returns them in sorted order, So taking the first will be the smallest/oldest.
`versions` does not document this behavior, but this is only used for testing so I don't mind depending on undocumented behavure.